File: TransCoder_tokenized_test_set_functions/java/POLICEMEN_CATCH_THIEVES.java
static int policeThief ( char arr [ ], int n, int k ) {
  int res = 0;
  ArrayList < Integer > thi = new ArrayList < Integer > ( );
  ArrayList < Integer > pol = new ArrayList < Integer > ( );
  for ( int i = 0;
  i < n;
  i ++ ) {
    if ( arr [ i ] == 'P' ) pol . add ( i );
    else if ( arr [ i ] == 'T' ) thi . add ( i );
  }
  int l = 0, r = 0;
  while ( l < thi . size ( ) && r < pol . size ( ) ) {
    if ( Math . abs ( thi . get ( l ) - pol . get ( r ) ) <= k ) {
      res ++;
      l ++;
      r ++;
    }
    else if ( thi . get ( l ) < pol . get ( r ) ) l ++;
    else r ++;
  }
  return res;
}

File: results/py/POLICEMEN_CATCH_THIEVES.cpp
int f_filled ( char arr [ ] , int n , int k ) {
  int i = 0 ;
  int l = 0 ;
  int r = 0 ;
  int res = 0 ;
  int thi [ n ] ;
  int pol [ n ] ;
  while ( i < n ) {
    if ( arr [ i ] == 'P' ) pol [ i ] = i ;
    else if ( arr [ i ] == 'T' ) thi [ i ] = i ;
    i ++ ;
  }
  while ( l < thi [ l ] && r < pol [ r ] ) {
    if ( ( abs ( thi [ l ] - pol [ r ] ) <= k ) || ( thi [ l ] < pol [ r ] ) ) {
      res ++ ;
      l ++ ;
      r ++ ;
    }
    else if ( thi [ l ] < pol [ r ] ) l ++ ;
    else r ++ ;
  }
  return res ;
}
